modules/cfe_testcase/src/time_conversion_test.c

```diff
@@ -0,0 +1,135 @@
+/*************************************************************************
+**
+**      GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**      Copyright (c) 2006-2021 United States Government as represented by
+**      the Administrator of the National Aeronautics and Space Administration.
+**      All Rights Reserved.
+**
+**      Licensed under the Apache License, Version 2.0 (the "License");
+**      you may not use this file except in compliance with the License.
+**      You may obtain a copy of the License at
+**
+**        http://www.apache.org/licenses/LICENSE-2.0
+**
+**      Unless required by applicable law or agreed to in writing, software
+**      distributed under the License is distributed on an "AS IS" BASIS,
+**      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**      See the License for the specific language governing permissions and
+**      limitations under the License.
+**
+** File: time_conversion_test.c
+**
+** Purpose:
+**   Functional test of basic Time Conversion APIs
+**
+*************************************************************************/
+
+/*
+ * Includes
+ */
+#include "cfe_test.h"
+
+void TestConvertMET2SCTime(void)
+{
+    UtPrintf("Testing: CFE_TIME_MET2SCTime");
+
+    /* Mission Elapsed Time */
+    CFE_TIME_SysTime_t MET;
+    /* MET + SCTF */
+    CFE_TIME_SysTime_t TAI;
+    /* MET + SCTF - Leap Seconds */
+    CFE_TIME_SysTime_t UTC;
+    /* Spacecraft Time */
+    CFE_TIME_SysTime_t SCTime;
+
+    OS_time_t          start;
+    OS_time_t          end;
+    OS_time_t          difference;
+
+    /* Print buffers */
+    char timeBuf1[sizeof("yyyy-ddd-hh:mm:ss.xxxxx_")];
+    char timeBuf2[sizeof("yyyy-ddd-hh:mm:ss.xxxxx_")];
+
+    OS_GetLocalTime(&start);
+
+    /* Get Times */
+    MET = CFE_TIME_GetMET();
+    TAI = CFE_TIME_GetTAI();
+    UTC = CFE_TIME_GetUTC();
+
+    OS_GetLocalTime(&end);
+
+    /* Convert - should produce a TAI or UTC at the moment of GetMET() */
+    SCTime = CFE_TIME_MET2SCTime(MET);
+
+    /* write Spacecraft Time into second buffer */
+    CFE_TIME_Print(timeBuf2, SCTime);
+
+    difference = OS_TimeSubtract(end, start);
+
+    /* Check conversion */
+#if (CFE_MISSION_TIME_CFG_DEFAULT_TAI == true)
+    /* SCTime is TAI format */
+
+    /* avoid unused compiler warning */
+    (void)UTC;
+
+    /* write TAI into first buffer */
+    CFE_TIME_Print(timeBuf1, TAI);
+
+    UtAssert_True(TimeInRange(SCTime, TAI, difference), "TAI (%s) = MET2SCTime (%s)", timeBuf1, timeBuf2);
```

Comment:
The print out is backwards here, should be  

UtAssert_True(TimeInRange(SCTime, TAI, difference), "MET2SCTime (%s) = TAI (%s)", timeBuf2, timeBuf1);